Mage/src/main/java/mage/abilities/decorator/ConditionalActivatedAbility.java

```diff
@@ -18,24 +18,36 @@ public class ConditionalActivatedAbility extends ActivatedAbilityImpl {
     private static final Effects emptyEffects = new Effects();
 
     private String ruleText = null;
+    private boolean showCondition = true;
 
     public ConditionalActivatedAbility(Effect effect, Cost cost, Condition condition) {
         this(Zone.BATTLEFIELD, effect, cost, condition);
     }
 
+    public ConditionalActivatedAbility(Effect effect, Cost cost, Condition condition, boolean showCondition) {
+        this(Zone.BATTLEFIELD, effect, cost, condition);
+        this.showCondition = showCondition;
+    }
+
     public ConditionalActivatedAbility(Zone zone, Effect effect, Cost cost, Condition condition) {
         super(zone, effect, cost);
         this.condition = condition;
     }
 
     public ConditionalActivatedAbility(Zone zone, Effect effect, Cost cost, Condition condition, String rule) {
+        this(zone, effect, cost, condition, rule, true);
+    }
+
+    public ConditionalActivatedAbility(Zone zone, Effect effect, Cost cost, Condition condition, String rule, boolean showCondition) {
         this(zone, effect, cost, condition);
         this.ruleText = rule;
+        this.showCondition = showCondition;
```

Comment:
For infrequently used params, don't clutter constructors. Instead add a chaining method.